### ISSUE TITLE
Item seems to affect the backward process of DataParallel

### ISSUE THREAD

**Issue by Richie-yan** [pytorch/pytorch#44103]
PyTorch :1.5.0
model : resnet50

Background:
I currently have two test environments, one is a physical machine with 8 gpus, and the other is a 4 gpus Docker on this physical machine.
Run the 4 gpus resnet50 model (using DataParallel) in these two environments. The code of the train part is as follows:
```
model.train()
for iters in range(1, train_interval + 1):
    data, target = train_dataloader.next()
    output = model(data)
    loss = loss_function(output, target)
    optimizer.zero_grad()
    loss.backward()
    optimizer.step()
    prec1, prec5 = accuracy(output, target, topk=(1, 5))
    Top1_err += 1 - prec1.item() / interval
    Top5_err += 1 - prec5.item() / interval
```

It is found that the training speed of running 4 gpus on a physical machine is 0.33 s/step, but the speed of running 4 gpus on Docker is 0.26 s/step. The speed difference between the two has reached more than 20%. After analysis, the main difference between the two is Reverse the gradient and parameter update, and find
After removing the item function, the speeds of the two are: physical machine: 0.274 s/step Docker: 0.264 s/step

Doubt:
1. Why does the item function cause such a big difference in training speed between the two?
2. It seems that the item function affects the process of reverse gradient calculation and parameter update?

**Comment by zou3519** [comment]
> 1. Why does the item function cause such a big difference in training speed between the two?

The item() function causes a device-to-host sync that may be impacting performance. To learn more, one can run a cuda profiler through your code (something like nvprof).

> It seems that the item function affects the process of reverse gradient calculation and parameter update?

What exactly is the behavior you are seeing here?

**Comment by Richie-yan** [comment]
Hi, @zou3519 
thank you for your reply
1. item has a synchronous effect, but what I don’t understand is why the effect of this function is very different in my two cases
2. The phenomenon I saw using nvprof is:
When the item function is called, the ncclReduceRingLLKernel call of the physical machine with 8 gpus accounts for a very high 17% of about 2ms, but the ncclReduceRingLLKernel call of Docker with 4 gpus accounts for only 4% of about 900us
The call of ncclReduce is called after the reverse gradient calculation, so I will observe that item affects the process of reverse gradient calculation and parameter update. When I remove the item function, the proportion of ncclReduceRingLLKernel in both cases is about 4%

**Comment by Richie-yan** [comment]
@zou3519 @ptrblck
Can you help answer this question?
